stan/math/prim/prob/von_mises_cdf.hpp

```diff
@@ -0,0 +1,172 @@
+ * Mises cdf for k > 50. In this regime, the von Mises cdf
+ * is well-approximated by a normal distribution.
+ */
+template <typename T_x, typename T_k>
+return_type_t<T_x, T_k> von_mises_cdf_normalapprox(const T_x& x, const T_k& k) {
+  using std::exp;
+  using std::sqrt;
+
+  const auto b = sqrt(2 / pi()) * exp(k) / modified_bessel_first_kind(0, k);
+  const auto z = b * sin(x / 2);
+  const double mu = 0;
+  const double sigma = 1;
+
+  return normal_cdf(z, mu, sigma);
+}
+
+/**
+ * This function calculates the cdf of the von Mises distribution in
+ * the case where the distribution has support on (-pi, pi) and
+ * has mean 0. If k is sufficiently small, this function approximates
+ * the cdf with a Gaussian. Otherwise, use the expansion from scipy.
+ */
+template <typename T_x, typename T_k>
+return_type_t<T_x, T_k> von_mises_cdf_centered(const T_x& x, const T_k& k) {
+  double ck = 50;
+  using return_t = return_type_t<T_x, T_k>;
+  return_t f;
+  if (k < 49) {
+    f = von_mises_cdf_series(x, k);
+    if (f < 0) {
+      f = 0.0;
+      return f;
+    }
+    if (f > 1) {
+      f = 1.0;
+      return f;
+    }
+    return f;
+  } else if (k < 50) {
+    f = (50.0 - k) * von_mises_cdf_series(x, 49.0)
+        + (k - 49.0) * von_mises_cdf_normalapprox(x, 50.0);
+    return f;
+  } else {
+    f = von_mises_cdf_normalapprox(x, k);
+    return f;
+  }
+}
+
+}  // namespace internal
+
+/** \ingroup prob_dists
+ * Calculates the cumulative distribution function of the von Mises
+ * distribution:
+ *
+ * \f$VonMisesCDF(x, mu, \kappa) = \frac{1}{2\pi I_0(\kappa)} \int_{-\mu-\pi}^x
+ * e^{\kappa cos(t - \mu)} dt\f$
+ *
+ * where
+ *
+ * \f$\mu \in \mathbb{R}\f$, \f$x \in (\mu - \pi, \mu + \pi)\f$, and \f$\kappa
+ * \in \mathbb{R}^+\f$.
+ *
+ * @param x A scalar variate
+ * @param mu The mean of the distribution
+ * @param k The inverse scale of the distriubtion
+ * @return The von Mises cdf evaluated at the specified arguments
+ * @tparam T_x Type of x
+ * @tparam T_mu Type of mean parameter
+ * @tparam T_k Type of inverse scale parameter
+ */
+template <typename T_x, typename T_mu, typename T_k>
+inline return_type_t<T_x, T_mu, T_k> von_mises_cdf(const T_x& x, const T_mu& mu,
+                                                   const T_k& k) {
+  static char const* function = "von_mises_cdf";
+  using return_t = return_type_t<T_x, T_mu, T_k>;
+  using internal::von_mises_cdf_centered;
+  const double pi = stan::math::pi();
+
+  check_not_nan(function, "Random variable", x);
+  check_not_nan(function, "Scale parameter", k);
+  check_not_nan(function, "Location parameter", mu);
+  check_positive(function, "Scale parameter", k);
+  check_less_or_equal(function, "Random variable", x, pi);
+  check_greater_or_equal(function, "Random variable", x, -pi);
+
+  // shift x so that mean is 0
+  return_t x2 = x - mu;
+
+  // x is on an interval (2*n*pi, (2*n + 1)*pi), move it to (-pi, pi)
+  x2 += pi;
+  const auto x_floor = floor(x2 / TWO_PI);
```

Comment:
Do we still need this modulus logic?

Reply:
unfortunately, I think we still do. I played around with it a bit in the hopes of deleting some of these lines, but I could not find a good solution.

Reply:
The comment is now wrong, though, and I'll fix that